app/models/katello/concerns/organization_extensions.rb

```diff
@@ -228,6 +228,13 @@ def expiring_subscriptions
           subscriptions.select(&:expiring_soon?)
         end
 
+        def clear_syspurpose_status
+          # rubocop:disable LineLength
+          host_purpose = HostStatus::Status.where(type: ['Katello::PurposeStatus','Katello::PurposeAddonsStatus','Katello::PurposeUsageStatus','Katello::PurposeRoleStatus','Katello::PurposeSlaStatus']).where('host_id in (?)', self.hosts.pluck(:id))
```

Comment:
This is looking really good. Now we see that the line is very long. We shouldn't bypass rubocop whenever possible, and this is very easy to work around in this case.

You can pull the list of types out of the query and put them somewhere else. Even just a variable above this query in the same method would do - but there's somewhere better: https://github.com/Katello/katello/blob/master/app/services/katello/host_status_manager.rb

Create a new constant there with the class constants - not strings - in an array. You can then reference that constant here and call to_s on its elements to get strings: `MY_CONSTANT.map(&:to_s)`. This achieves not only addressing the rubocop lint, but now all the purpose statuses are grouped together which could be beneficial in the future for new code that might need to know about them. We can change the value of that constant in one place and the code referencing it should continue to work.

Here's a challenge that I would _highly_ recommend: before changing this code, write a passing unit test that has assertions for all the intended behavior of this method. Then go about refactoring as I suggested. If the tests pass then you got it working. If not, you created a bug! Adding tests at the very end of a code change is much less useful than writing them at the start (you know I had to sneak that in there). This is the perfect opportunity to try out some TDD